app/main.py

```diff
@@ -1,13 +1,23 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"}
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path, "status": "failed"}
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [{
+        "errors": [],
+        "path": file_name,
+        "status": "passed"
+    } if linter_report[file_name] == []
+        else format_single_linter_file(file_name, linter_report[file_name])
```

Comment:
The conditional expression here checks if `linter_report[file_name]` is an empty list to determine the status of the file. Ensure that `linter_report[file_name]` is always a list, even if there are no errors, to avoid potential issues. If `linter_report[file_name]` can be `None` or another type, this could lead to unexpected behavior.